source/tutorials/learning-journey/packaging-existing-software.md

```diff
@@ -300,8 +313,34 @@ stdenv.mkDerivation {
 }
 ```
 
+Now is a good time to change the `version` string also.
+
+To avoid repeating yourself, and to make later updates to the package a little bit easier, you should set the `version` to `0.5` and then use [string interpolation](https://nix.dev/tutorials/first-steps/nix-language#string-interpolation) in the `rev` attribute passed to `fetchFromGitHub`.
+
+For `version` to be accessible to `rev`, the attribute set passed to `stdenv.mkDerivation` must be [*recursive*](https://nix.dev/tutorials/first-steps/nix-language#recursive-attribute-set-rec), so you should also add the `rec` keyword to avoid seeing `error: undefined variable 'version':
```

Comment:
It may be better practice to use a `let` binding instead of `rec` since `rec` can make it hard to trace data flow.

Reply:
Normally I would, but last time I submitted, I was [told not to use `let`](https://github.com/NixOS/nixpkgs/pull/239035#discussion_r1240091786) :upside_down_face:

Reply:
Ah, so we're actually supposed to do [this](https://nixos.org/manual/nixpkgs/stable/#mkderivation-recursive-attributes)

Reply:
Sigh

Maybe you're "supposed" to do the last one, but that's so much more complicated than the other solutions that I'm inclined to say just throw your hands up and use `rec`.

Reply:
`finalAttrs` is absolutely the right way to do it and severely underdocumented. In suppose has to be taught somewhat separately because it's tied into overrides, but for now we could do a compromise and use either let or rec with a note that there also exist finalAttrs, and then later reorder the tutorials to start with consuming existing packages. Since I took it upon me to extend the curriculum running into these issues is valuable input.